pkg/networkservice/common/heal/client.go

```diff
@@ -62,7 +63,7 @@ func (h *healClient) Request(ctx context.Context, request *networkservice.Networ
 	}
 	cc, ccLoaded := clientconn.Load(ctx)
 	if ccLoaded {
-		cancelEventLoop, eventLoopErr := newEventLoop(h.chainCtx, begin.FromContext(ctx), cc, conn, h.livelinessCheck)
+		cancelEventLoop, eventLoopErr := newEventLoop(h.chainCtx, begin.FromContext(ctx), cc, conn, h.livelinessCheck, log.FromContext(ctx))
```

Comment:
```suggestion
		cancelEventLoop, eventLoopErr := newEventLoop(extend.WithValuesFromContext(h.chainCtx, ctx), cc, conn, h.livelinessCheck)
```

Then you can get logger and begin from the ctx:

```
b := begin.FromContext(ctx)
logger := log.FromContext(ctx).WithField(...)
```

Reply:
thank you, fixed